### ISSUE TITLE
tsf_voice_kill not being called for all Voices

### ISSUE THREAD

**Issue by adrianmcroft** [schellingb/TinySoundFont#34]
It is possible for voice->playingPreset to be set to -1 during tsf_note_on before completion. This is caused by tsf_voice_kill altering the voice during render (assuming a multithreaded call) as soon as voice initialisation begins. This means that the voice pool grows over time. I've added an active flag (in my local code) around tsf_voice to protect it until it is fully set.

**Comment by schellingb** [comment]
The [documentation ](../blob/master/tsf.h#L121) briefly mentions that TSF itself is not thread safe and that if tsf_render* functions and tsf_note* functions are running on separate threads the user is responsible for concurrency control.

See [example2.c](../blob/master/examples/example2.c#L9) for how it uses a mutex to avoid conflicts.

For Bela it seems multi threading is done with standard posix threads so look into [pthread_mutex_t](https://www.thegeekstuff.com/2012/05/c-mutex-examples/).

**Comment by adrianmcroft** [comment]
Ah no absolutely, its not a issue with TSF (although here it is in Issues) - I just raised it as a hurdle for myself to overcome and the primary cause of my problems is my note-stealing code. 

I've implemented 'lockless' queues for Note On and Offs and now the flow is serialised, I can loop the midi queue during Render without worry about racing. I've left it overnight with both my Windows Vst and Bela versions playing the nastiest classical Midi Files I could find, with no dropouts and low CPU (48 voice max and Reverb on). 

Many thanks as usual Bernhard....
